### ISSUE TITLE
Create a Notification System

### ISSUE THREAD

**Issue by loorm** [citizenos/citizenos-fe-old#718]
**What is the user story?**
As a registered user of Citizen OS, who is a member of at least one topic or group, I want to receive notifications of activities without going to Citizen OS app. Activities in this case means actions by other users or automatic scripts of Citizen OS, that influence content related to me. 

Full user epic with comments here: https://docs.google.com/document/d/1k52Sw-ShqI9p3Vh_D_zZwRjbk8KAgeAWA-4O1XtSsec/edit?usp=sharing

**What is the requested feature?**

- I want to be able to switch all notifications on and off for a single topic with 1 click. Also for a single group with 1 click. This should default to all actions, by e-mail and daily digest.
- I want to be able to choose, which notifications I will be receiving, when and which channels should those notifications be sent to (see table below).
- I want an option to turn all notifications on or off in bulk.
- I want an option, where I can either exclude or include notifications about my own actions.
- By default, I want all notifications to be turned off.
- I want e-mail and SMS notifications to contain a non-changing marker, that would enable me to auto-filter the notifications.
- I want the notification to state in a simple format and very clear terms (i.e. in case of e-mail, I want it to be both on the subject line and immediately visible in the e-mail body), what activity am I being notified of, when (time + time zone) was the action performed and who performed it.
- I want notifications to be clickable, so I can go directly from the notification to the relevant content in Citizen OS app.
- I want the notification to contain a link or explanation to the configuration page, where I can change notification settings.
- I want public topics to have public notifications, published in a machine-readable format, so that I can write a script to import these notifications into a FB group, on my website or elsewhere.

Below are the choices I want to be able to make. Even though the table below also shows in-ap activity feed, I do not want this to be configurable. I want to simply see it here for informative purposes. The choices should be allowed in parallel (i.e. I am able to choose both SMS notification and e-mail summary notification for the same action)

![notification system](https://user-images.githubusercontent.com/22661743/118123070-61456380-b3fc-11eb-9148-afd1530b8daa.jpg)

**Additional information.**
This needs to be future proof - when we add new modules (like ideation or private messaging), notification options need to expand.

Need to add phone number field to account, in order to support SMS sending. When switching SMS option on, system needs to check for and notify absence of phone number in profile.

NB! There is a previous investigation issue about push notifications here https://github.com/citizenos/citizenos-fe/issues/287

**Comment by loorm** [comment]
Prioritizing for development outside of triage. Agreed this with @ilmartyrk. Reason: we have a temporary developer, who can start working on the Notification System immediately.

**Comment by loorm** [comment]
@kevincrepin This needs a lot of input from you:

- [ ] We need to put this notification configuration page somewhere. And design it, of course.
- [ ] We need a button or smth in every topic and group, that can be used as a quick-switch for notifications (first bullet-point in user epic)
- [ ] We need some simple layout for e-mail notifications.
- [ ] Notification texts can largely be perused from activity feed strings.
- [ ] This is also going to create quite a few new strings for l10n @helenkask, please mention to Sepp.

**Comment by ilmartyrk** [comment]
Created a document in Citizen OS for overview of different components that backend should have. This is a task list for @el-tu to start going through: 
- [ ] create the structure for notification object (look for standard or create own logic) 
- [ ] check how FE handles activity-feed (translation keys, getting links etc.)
- [ ] linking activities with notifications and user preferences

**Comment by BeccaMelhuish** [comment]
Just going through the FAQs and which solutions we might be able to start testing with users soon, and as this relates to FAQ 8: "In what situations does the Citizen OS platform send out automatic email notifications to users?" 

So @ilmartyrk please keep me posted when there is something concrete to be tested, such as a design and a way of navigating to the navigation configuration page (e.g. where the user will find it), etc. and I can help to work out how we go about testing it (with some help from @kevincrepin hopefully :))

**Comment by KatiVellak** [comment]
@ilmartyrk and @el-tu kindly note that by default, all notifications must be turned off. User can actively opt-in to messages (not actively given the chance to opt out).

**Comment by loorm** [comment]
> @ilmartyrk and @el-tu kindly note that by default, all notifications must be turned off. User can actively opt-in to messages (not actively given the chance to opt out).

Good point! This means, that we need to also build active reminders for users to switch on their notifications.

**Comment by ilmartyrk** [comment]
Checklist of related issues and tasks:
- [ ] related issue https://github.com/citizenos/citizenos-fe/issues/111

**Comment by kevincrepin** [comment]
@anettlinno is this something we still want to put an effort in for this version of the platform?

**Comment by anettlinno** [comment]
> @anettlinno is this something we still want to put an effort in for this version of the platform?

Notification system is getting more and more relevant and many issues are "on-hold" until we start developing this (please check the Big Picture board where related issues are linked: https://github.com/orgs/citizenos/projects/1). On our last Triage meeting we discussed this with Kati and @ilmartyrk and agreed that we should start developing this asap. So, it you, @kevincrepin and @tiblu think that it is doable before we kick-off the redesign project then I would say YES, let´s do it!

**Comment by TiinaDUX** [comment]
Hey All

Glad to share first solutions for notifications topic, please take a look. We worked on: 1. Notifications settings in account view 2. Notification settings in topic view and 3. Example about e-mail template.

Figma link: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1034%3A25417

Welcome to feedback!

**Comment by TiinaDUX** [comment]
Hey All

We have found first solutions for notification topic. Welcome to check!

**1) Notifications/ account based**
Since notification settings should be flexible and already include a lot of content, it would be more convenient to put it on a separate page rather than a popup. The user expects to find the settings in the section of the personal account, so we want to propose a new kind of personal account with the same logic but a separate page. 
Link:
https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1921%3A32446

Here is also extra screen to show how the other account tabs could look since it's not a popup anymore. 
Link: 
https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1960%3A33161

**2) Notifications/ topic based**
To configure notifications of a separate topic we added an action that opens a popup where the user can select communication channels and what notifications to receive. 
Link:
https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1923%3A38612

**3) E-mail example**
In the letter to the mail, there will be a link to the topic and a link opening a popup with the notification settings for this topic
Link:
https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=1921%3A38172

**Comment by anettlinno** [comment]
Thanks @TiinaDUX for the designs! Here are my thoughts: 

1) Design looks clean and beautiful -- **love it!** As @KatiVellak commented then by default all notifications must be turned OFF and users should have an opportunity to turn them ON from the Notification Management tab. 

- Regarding the "Important updates in App". As we currently do not have this page in our App, then we need to figure out where we hold latest updates information (if I remember correctly then we discussed on our UX/UI meeting that this could be held in the main menu where we currently have "About" and "Help"). This could also be held on our web site. How does User get the latest updates?  We also need design for Updates page. I´m going to create separate issue for that.
- Regarding in App Activity Feed. Could these be ON by default, @KatiVellak? As in App notifications are passive and User must click on the Activity Feed to see those.
- Regarding the SMS. We don´t currently ask users phone number when User registers and I personally think it´s OK not to ask it then. BUT if User chooses SMS as one of the notification channels, then we should get her phone number. How? With the pop-up window? Should we direct her on My Account to enter phone number? Any other ideas? :)
- I have some questions about Push, but as we have separate issue #287 for that, I will leave my questions and comments there. 

2) Topic notifications are harder than the account based notifications, right? For example if as a User I will switch OFF "All actions in my Topics" but will switch ON "Voting" notification under Topic notifications (and choose the channel) then I will receive notifications only about the votings on this topic, right? What if as a User I have switched OFF all my notifications and will choose "Voting" but do not choose any channel? I think User should be directed then to choose the channel as well. 

3) E-mail design looks beautiful and includes all the relevant information IMHO.

I´m thinking that if as a user I have switched ON notifications in 3 groups and 5 topics and in one moment I feel that I don´t want to receive those any more as I´m overwhelmed, then there could be a place where can I see all the Groups and Topics where I have changed the notifications and easily switch them all OFF. Or at least manage those in one place. So, we could provide an overview about all the topics and groups where User has changed the notifications OR provide User an opportunity to switch OFF Topic and Group notifications from one place. In the last case then the notifications under the Notification Management Tab should be harder than Group´s or Topic´s notifications. 

Looking forward to your ideas @tiblu @ilmartyrk @loorm

**Comment by anettlinno** [comment]
In issue #289 we have planned to send notification for the admin user if somebody requests access to the topic. Do you think @TiinaDUX this should be separate thing or could go under the User Administration for example? 

I´m also thinking that we should have explanatory texts somewhere (maybe in the Help Center?), where User can read what each action consists of (for example "All actions in my topic" consists of notifying if another user have edited the text, left an argument, changed topic´s privacy etc). This will help to make more informative decision because User can then consider if this is too much for her or not.

**Comment by loorm** [comment]
> 2\. Topic notifications are harder than the account based notifications, right? For example if as a User I will switch OFF "All actions in my Topics" but will switch ON "Voting" notification under Topic notifications (and choose the channel) then I will receive notifications only about the votings on this topic, right? What if as a User I have switched OFF all my notifications and will choose "Voting" but do not choose any channel? I think User should be directed then to choose the channel as well.

Because of clashes like this, I would not have the separate notification choices for groups/topics at all. I would just have the general notification settings, where you can choose channels and events. I think that looks great. 

For each group/topic I would only have notifications on-off switch. When "on", the general notification rules are followed. When "off", you don't get any. Except the admin or moderator notifications, these are separate and can't be opted out of, otherwise the system doesn't work. 

This would also get rid of the need to show the user a list of topics / groups, where they have changed their notifications. 

I propose to create a simpler solution. Dealing with all the clashing conditions will otherwise be a pain.

**Comment by anettlinno** [comment]
> > 2. Topic notifications are harder than the account based notifications, right? For example if as a User I will switch OFF "All actions in my Topics" but will switch ON "Voting" notification under Topic notifications (and choose the channel) then I will receive notifications only about the votings on this topic, right? What if as a User I have switched OFF all my notifications and will choose "Voting" but do not choose any channel? I think User should be directed then to choose the channel as well.
> 
> Because of clashes like this, I would not have the separate notification choices for groups/topics at all. I would just have the general notification settings, where you can choose channels and events. I think that looks great.
> 
> For each group/topic I would only have notifications on-off switch. When "on", the general notification rules are followed. When "off", you don't get any. Except the admin or moderator notifications, these are separate and can't be opted out of, otherwise the system doesn't work.
> 
> This would also get rid of the need to show the user a list of topics / groups, where they have changed their notifications.
> 
> I propose to create a simpler solution. Dealing with all the clashing conditions will otherwise be a pain.

Very much agree with the proposal to create a simple solution and leave out the Groups / topics notification choices!

**Comment by anettlinno** [comment]
Triage 54. Discussed and decided to create an MVP version of notification system in the first phase. This would include: 

- Creating separate "My account" page with notification window (as suggested by @TiinaDUX in her design above);
- Notifications panel include two options: "Activity feed" and "Email" where activity feed is opt-in by default and email is opt-out by default. Technically this will be developed so that later it would be easy to integrate other channels (SMS, Slack) if needed;
- Email design is beautiful and texts should be reusable for most (if not all) of the actions. "Turn off notifications" directs user on the notifications window where User can easily opt-out from all notifications. Emails should be user tested before we launch the notification system. 

Est. dev. time 32h. Sending to development.

**Comment by kevincrepin** [comment]
Hey,

We've added a reference of how the dialog window can look like for the topic notifications if we keep it simple without any specific settings but just turning on/off: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform?node-id=2103%3A33819

If I understand correctly then the rest of the designs are still relevant, but some parts will not be developed yet (ie. different channels)?

**Comment by anettlinno** [comment]
Hi @kevincrepin,

In the first iteration, we discussed, that we implement only separate My Account page and setting up e-mail notifications. Our main concern with adding separate dialog window under topic/group is that when User wants to switch off all notifications then where can she do it? And how intuitive is it to find this place for the User? 

If the topic notification would only have ON/OFF button and only those notifications which are switched on under the "master" notification window would apply, then I think we could include this one into the first iteration as well. Then all the notifications (incl. all topics and groups) should be controlled from the notifications window and it is harder than group/topic notifications. 

My question is that if User clicks on the "Turn off notifications for this topic" button in her e-mail then will she be directed on the notifications window or topic/group notifications?

@ilmartyrk what do you think about adding group/topic notifications dialog window in the way as described above?

**Comment by kevincrepin** [comment]
Yes, I agree that at this point it should be so that there's only an on/off switch per topic (assuming we want this level of flexibility). So that they would easily find it we would suggest having it literally/separately in the actions menu:
![Screenshot 2021-12-15 at 11 08 19](https://user-images.githubusercontent.com/26792281/146157066-9ce8dbc1-7eb5-4b34-ad5d-9ce9c69d1af7.png)

**Comment by anettlinno** [comment]
Triage 55. Estimated dev. time. 2 weeks.
Before launching this issue, we do the user tests @BeccaMelhuish 

Developing API back end together with #317

**Comment by anettlinno** [comment]
Tested notification system in test environment. My feedback:

- "Topic notification" button should opt-in automatically after User marks any tick, because current flow (opt-in "Topic notifications" button and then tick) is not intuitive;
- If I opt-out from the notifications then I would like that system would empty tick boxes as well, as in current version, I´m not sure, if my selections are opted-out or not;
- Setting topic notifications information should also be displayed in "My Activity" feed.
- I opted-in all notifications, but I haven´t received any email about topic content change.

AND there should be a way where and how I can opt-out all notifications.

**Comment by ilmartyrk** [comment]
@anettlinno for better UX should we then remove the toggle button and just keep checkboxes under topic notification settings? I will add notification settings activity to activity feed and will also add notification for content changes. Also by opt-out all notifications you mean that we need a toggle on/off notifications for all topics in one place?

**Comment by anettlinno** [comment]
> @anettlinno for better UX should we then remove the toggle button and just keep checkboxes under topic notification settings?

I think it might work, @ilmartyrk if to remove the toggle button at all. 

> I will add notification settings activity to activity feed and will also add notification for content changes. 

Thanks!

> Also by opt-out all notifications you mean that we need a toggle on/off notifications for all topics in one place?

Yes, I meant that we should have one place where people can see in which topics they have opted-in notifications and they should have an opportunity to opt them all out.

**Comment by ilmartyrk** [comment]
@anettlinno I believe that this general notifications for all topics is out of our mvp scope. And we would need a UX/UI design for that. 

Stll I suggest adding list of topics as it is quite easy to query and then adding button that opens the same settings popup as in current topic view. Also I could use the toggle button there to make turning all notifications on/off for a topic faster

**Comment by anettlinno** [comment]
@ilmartyrk you are right -- as we discussed then general notification will not be included in mvp version. 

> Stll I suggest adding list of topics as it is quite easy to query and then adding button that opens the same settings popup as in current topic view. Also I could use the toggle button there to make turning all notifications on/off for a topic faster

I think this is good idea! This list will be then under "My account" and "Notifications"?

**Comment by anettlinno** [comment]
Tested. Here is what I experienced: 

- As a User I don´t understand what is the "Reports". What kind of reports?
- "Follow up events" is not noticable;
- I received the same message for changing the content, sending topic to voting and adding hashtab: "_User Anett Linno has updated a field in topic Notification testing_" Maybe we should create different texts so that Users could understand better what has changed?
- After switching off all notifications, I still received notification email after sending the Topic to follow-up and posting an update (email texts are not clear for me to understand what kind of updates were done: _User Anett Linno has created an update on topic Notification testing_ AND _User Anett Linno has updated a field in topic Notification testing_)

**Comment by anettlinno** [comment]
In notification email we have this text: _Thank you! Together we can maintain a high quality of content in Citizen OS user community._
![Email](https://user-images.githubusercontent.com/45175679/173032272-4f0b6bbf-be15-427a-84bb-58ff42fad733.png)



My first feeling after reading this message was that I´m forced to be part of some kind of community I don´t know anything about... I think we should change this text for something more neutral :) 

And actually review all notification email texts.

**Comment by BeccaMelhuish** [comment]
Hey @ilmartyrk and @anettlinno! 

Tried to do some testing but couldn't work out how it's meant to work.. Here's what I tried and what happened: 

First I went to Account (as I remembered talk of that being where it would be), and just found this odd empty page, with just "delete account" on it. (Should that not only be on Profile and Account?) I tried clicking around to see if anything happened - on the title and the empty line above "Save" but nothing did. (The heading turns bold and the cursor turns to a hand though, as if it should - is it meant to do that?). It looked unfinished to me, like it's not loaded fully or something. 

![image](https://user-images.githubusercontent.com/61320526/173362890-cfebbd42-7002-484d-b1bb-c97d6f3bcce4.png)

Then I tried here below and had a lot of trouble getting the bar to expand down (sometimes it expanded when clicking on the bar, sometimes nothing happened, same with the "+" sign. About 1 in 8 times it worked, then rest were just non-registered clicks (on Chrome)): 

![image](https://user-images.githubusercontent.com/61320526/173363022-feb48afb-b62b-4e31-a273-d9f6a489cd09.png)

Then I tried the settings of the topic itself but nothing here either: 

![image](https://user-images.githubusercontent.com/61320526/173363335-64a7ff4b-1c02-4a49-8aff-945b93821b08.png)

Then I tried the main activity feed too just in case it was there. 

Then gave up and came here to write this. And now from looking through the thread a bit and seeing Anett's screenshot above I see that the notifications appear in the Actions menu when you are inside the topic itself, rather than in the dashboard. So I'll now go ahead and test them, but just sharing this as two first issues: 
(1) Not very easy to find 
(2) When there are no notifications set, the empty notifications tab (first screenshot above) is quite odd and confusing
((3) And a third side point - issues with the dashboard dropdown bars, but not so relevant to this thread I guess!).

**Comment by oksks** [comment]
Heya! Found an AF string that's related to this ticket here according to @ilmartyrk - needs being looked at:

<img width="784" alt="Screenshot 2022-06-14 at 21 49 44" src="https://user-images.githubusercontent.com/34126353/173667391-4424176d-0356-4c59-80f1-f5744eaadebf.png">

**Comment by BeccaMelhuish** [comment]
To be fair, re my "(1) Not very easy to find" above, it was a bit of an artificial test as I went onto COS with the express purpose of "testing the notification system" which no-one would ever do (so probably they would be more likely to find it within the topic). But still think there should be a way to edit the settings via the dashboard like the rest of the topic settings.

**Comment by BeccaMelhuish** [comment]
Some other points that I noted down during the discussion:

- All comes in the same thread not specifying which topic in the subject heading. If several topics were happening at once, all the notifications would be jumbled up. Is it possible to make it work like Google doc comment/suggestion notifications, so the subject heading of the email says the name of the topic, and then they are all separated into email threads just for that topic..  
- Related to the above, it seems to start a new thread each day, even if the topic is the same. I.e. All my notifications from yesterday are in one email thread together, but this morning Anett replied to something and it started a new thread. Would be better if all in one I think? (Though it could get extremely long..) 
- It gives me notifications for myself, which seems a bit unnecessary/annoying - when you think there's something of interest, go into the email to read the notification and find it's just telling you what you just did.. Then filtering through them all to see if there's one from someone else or they're all from you.. 
- It seems to notify me when someone replies to any argument, but it would be nice to be able set it to only show me replies to my arguments, or to any argument which I am "engaged" in (i.e. I have previously replied to). As if it tells me every time anyone replies to any argument in there, if there are a lot of arguments and replies this could get overwhelming (am of the long thinking 34 Islands project topics, for example). 
- It doesn't seem to notify when someone likes/dislikes something I wrote. I think I would want to be notified of that - it's easy to miss in the discussion. 
- If you edit the notifications then click "X" rather than "Save" (which I nearly did once then caught myself) it would be good to ask "Close without saving?" to be sure that's what they meant to do. 
![image](https://user-images.githubusercontent.com/61320526/173809611-bb947a3f-a686-4f1d-bca3-bbec6c682db8.png)
- Just a misbehaving comma here: 
![image](https://user-images.githubusercontent.com/61320526/173808766-de5425b7-90e5-414b-9744-33d090d3f19a.png)
- Wondered what "Reports" meant here - could be clearer?
![image](https://user-images.githubusercontent.com/61320526/173808896-d31ad685-58d3-4478-96a9-de017e13ad18.png)
- This is a bit weird here:
![image](https://user-images.githubusercontent.com/61320526/173809950-32a1623d-e7fb-46b2-9703-6d15dd32b904.png)
- Maybe could be "Edit notifications for this topic" instead? 
![image](https://user-images.githubusercontent.com/61320526/173811561-610fae6f-b4fe-43b3-93bc-91c038fc60a0.png)
- Quite often the notification would be hidden and I'd have to click on the three little dots to see it, not sure if there's a way to avoid that, so it always appears? 
![image](https://user-images.githubusercontent.com/61320526/173811965-8ecea7de-0d16-4220-96d7-74b1672c5eb7.png)

**Comment by BeccaMelhuish** [comment]
Also a grammar point, should be "A user" here.

But I also wonder if everyone wants to know every time someone votes. Especially if it's a big group.. Obviously they could turn it off, but then they would lose notifications for anything to do with voting, presumably, so including the vote ending, or being extended, or such like? 

![image](https://user-images.githubusercontent.com/61320526/173887251-ec982801-1050-4208-a65a-2d44a1586858.png)

**Comment by ilmartyrk** [comment]
So  I am taking out some development points and ignoring the grammar things as they or completely different tasks from my developer point of view.
1) We should not notify users about their own actions. - I agree, this is actually weird and annoying. @anettlinno what do you think?
2) Add topic title to e-mail subject
3) Reports should be clearer, it actually currently notifies if the topic has been reported by user

**Comment by BeccaMelhuish** [comment]
One little bug I just spotted, for the second project Sara shared, it missed off the name of the topic when Oks joined: 

![image](https://user-images.githubusercontent.com/61320526/174033470-fc083dd6-76d0-4e8d-abc4-f09bba661e37.png)

**Comment by BeccaMelhuish** [comment]
@ilmartyrk I think the "Close without saving?" message would be quite an important one from a UX point of view too, as a user could think they have changed the notifications, but they actually haven't (and then get annoyed that it didn't work and they are still getting them).

**Comment by BeccaMelhuish** [comment]
And if there was a way to make it so that there is a difference between "User X has replied to **an** argument" and "User X has replied to **your** argument" this would also be a pretty beneficial development to keep the user happy :) 

Though then there are complications with what if it is a reply to a reply and not to an argument? I've not got a good grasp currently of what happens when someone replies to your reply on someone else's argument - does it send you a "X has replied to your reply on an argument" message or what kind of message do you get? So if for example Sara reply to my reply to Ilmar's argument, is the message very generic like "Sara has replied to an argument"? As if so I think users would want it to be more specific - as it a direct "(at)Becca" message, not just one of (potentially) many other replies to (potentially) many other arguments.

**Comment by BeccaMelhuish** [comment]
Also @ssin1901 and @anettlinno, what do you think if we include Kevin in user testing this too? Perhaps Sara and him have a discussion on a topic and he reviews the notifications too? As I am just giving my hunches here from a UX point of view, but he is our UX expert :)

**Comment by BeccaMelhuish** [comment]
A few other things: I just logged back in and was looking at the notifications and it seems to have "forgotten" that I had them all ticked already! Just 10 minutes ago I was receiving notifications, and I hadn't changed it since... 

![image](https://user-images.githubusercontent.com/61320526/174041300-073730c3-13e9-477f-a017-90041c6c8de9.png)

OK and then I just clicked "Close" without ticking anything and tested adding another argument, and I still got a notification - something seems to be quite broken there! After receiving this notification I went back to check that the notification settings are all still unticked and they are. 

![image](https://user-images.githubusercontent.com/61320526/174042499-adb0aeeb-8962-41d3-b338-4513685f2aee.png)

**Comment by BeccaMelhuish** [comment]
Also, (why I discovered the above) I was looking for the option to turn off all notifications that say "User X has joined this topic" as I thought if it was an organisation with 100 people for example, or more, I wouldn't want 100 joining notifications muddled up with all the more important/interesting/relevant things. But I couldn't see which tick box this would be. Maybe it should be a separate tick box on it's own, so you could just stop getting those but still get all the rest?

**Comment by ssin1901** [comment]
@ilmartyrk don‘t know if it is on purpose, but when you toggle Topic Discussion (arguments, replies) on and off you don‘t get notified about Topic Discussions that were held while you had Topic Discussions off. 

It might be a UX thing – I am also not sure I would want to get a flood of notifications when I toggle discussion back on.

**Comment by ssin1901** [comment]
@ilmartyrk as Becca mentioned, there is an inconsistency in the two topics that is very inconvenient for me as a topic owner.

In the first topic, I see the name of the topic attached to the user who has joined the topic

![4451E4F2-E850-49AA-B16E-E9BFB6B6F73A](https://user-images.githubusercontent.com/72747107/174069708-6acfed44-86d6-4160-8044-1ff6951880b1.jpeg)

In the second topic I do not see it

![C73A5AAF-318D-4076-8342-A5C261F0E3CC](https://user-images.githubusercontent.com/72747107/174069894-c03096f1-7bdf-4c2d-a6cd-a45a5608ada1.jpeg)

I have no way of knowing which topic of mine they actually joined (I have two).

**Comment by oksks** [comment]
In notification emails, we're currently offering to **"Turn off notifications for this topic"** towards the end of the email. 

User subscribes to receiving notifications because he/she considers them important, however after receiving so many notification emails recently, the button I'm really missing in place of the current one is: **"Receiving too many notifications? See ways to group them."**

And then offering periodical summaries - send notifications immediately; or group all notifications into an email once a day, once a week, etc. Wdyt?

**Comment by anettlinno** [comment]
To sum it up and decide what to take in notifications systems MVP version and what to leave for the future developments. 

1) Topics list under the "My account" and "Notifications".
- User can then see all the topics where notifications are turned on and by clicking on the topic, she can opt-in or opt-out the notifications (I think that the "Set topic notifications" window will pop-up for changing the notifications, right @ilmartyrk?). 
- There is also an opportunity to turn off all notification separately topic by topic AND all notifications in all topics. The same view is where the email link "Turn off all the notifications" directs Users
- If User doesn´t have any notifications turned on, then what do we show them on the empty page @kevincrepin 

<img width="825" alt="My Account view" src="https://user-images.githubusercontent.com/45175679/174610315-c5886a7b-34df-4bcc-8e8a-6544af1b1085.png">

@kevincrepin we would need the UX/UI design for this tab as currently we have different view for "My account" but this has changed (designs: https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform-1.0?node-id=2103%3A33819). 

2) "Set topic notifications" from the topic Options:

<img width="359" alt="Set notifications" src="https://user-images.githubusercontent.com/45175679/174613493-03d78a16-733b-4a3b-ba0d-9366be9c156a.png">

What needs fixing: 

- Change the explanatory text by showing the topic´s name: "These settings will relate only to the topic @:TXT_TOPIC_"
- Opting out all notifications needs further testing and improvements as currently User still receives emails after switching off all notifications. Needs testing by @ssin1901 
- Move "Reports" under the "Topic updates" (_Topic updates (content, status, categories, reports etc_)"
- "Follow-up" events needs testing @ssin1901 (is it noticeable or not)
- Notify when someone likes/dislikes something User wrote. Add "Reactions" on Topic discussion (_Topic discussion (arguments, replies, **reactions_**))

3) "Notification from your Citizen OS account" email

<img width="454" alt="E-mail" src="https://user-images.githubusercontent.com/45175679/174746268-0f52d635-8682-4717-b7e2-367c4a6547bf.png">

What needs fixing: 

- Do not notify users about their own actions;
- Adding topic name in the email subject line. @ilmartyrk will check if there is character limit for the subject line, if yes, then we´ll add three dots... 
- User should have an opportunity to change topic notifications options and turn them all off from one place. For that we should have two options: 1) "Manage notification settings for this topic" (directs User to "Set topic notification" view) AND 2) Manage all notifications" (directs User to "My account" "Notifications" view
- E-mail texts will be changed after everything else works

Additional: 

- User with reading and editing rights cannot currently see the "Set topic notifications" option in Actions -- add "Set topic notifications" as an option for them as well
- Add "Topic notifications" under the General Information and "Topic actions"
<img width="544" alt="General info" src="https://user-images.githubusercontent.com/45175679/174752584-6cc36a7b-dd3c-4a24-b25f-eb49337fa554.png">

- Needs fixing: https://github.com/citizenos/citizenos-fe/issues/718#issuecomment-1155603310
- Please change "Back" button name to "Cancel"

**Comment by BeccaMelhuish** [comment]
@anettlinno should I say if some problems I found/UX improvement suggestions I made aren't in the summary list (or anyone else if they notice things they feel are important not there)? Not sure if they just got missed accidentally or was a decision to not address them yet..

**Comment by anettlinno** [comment]
> @anettlinno should I say if some problems I found/UX improvement suggestions I made aren't in the summary list (or anyone else if they notice things they feel are important not there)? Not sure if they just got missed accidentally or was a decision to not address them yet..

@BeccaMelhuish I´ll explain in separate posts why we left out some of them. If there is anything missing after that, please let know :) 

Regarding the notifications flow that you described [here ](https://github.com/citizenos/citizenos-fe/issues/718#issuecomment-1153917258)@BeccaMelhuish, we agreed that Sara @ssin1901 will conduct the user tests to understand which are the most common places where users search notifications on the platform. After we have that information, we can enhance notifications settings.

**Comment by anettlinno** [comment]
> * Related to the above, it seems to start a new thread each day, even if the topic is the same. I.e. All my notifications from yesterday are in one email thread together, but this morning Anett replied to something and it started a new thread. Would be better if all in one I think? (Though it could get extremely long..)

This is something that we cannot change as Gmail is grouping external email threads like that (starting new thread each day) and grouping only replied emails into one email feed. 

> * It seems to notify me when someone replies to any argument, but it would be nice to be able set it to only show me replies to my arguments, or to any argument which I am "engaged" in (i.e. I have previously replied to). As if it tells me every time anyone replies to any argument in there, if there are a lot of arguments and replies this could get overwhelming (am of the long thinking 34 Islands project topics, for example).

In this MVP version User can opt-in and opt-out all notifications in argumentary (arguments, reply, reactions). In later versions we might provide more opportunities (as described above) if there is a need for that, but our current aim is to learn about Users needs and behavior and then make enhancements according to that :) 

> * If you edit the notifications then click "X" rather than "Save" (which I nearly did once then caught myself) it would be good to ask "Close without saving?" to be sure that's what they meant to do.

We will change the "Back" button to "Cancel" button which means that if User wants to cancel her activities, she can do that by clicking on the button. For keeping the "X" consistency on our platform, we will not add the "Close without saving" question there now, because then we should change it everywhere. BUT we might want to rethink the whole "X" button system on our platform and talked to @ssin1901 to test this also with users.

>   ![image](https://user-images.githubusercontent.com/61320526/173809611-bb947a3f-a686-4f1d-bca3-bbec6c682db8.png)

> * Quite often the notification would be hidden and I'd have to click on the three little dots to see it, not sure if there's a way to avoid that, so it always appears?
>   ![image](https://user-images.githubusercontent.com/61320526/173811965-8ecea7de-0d16-4220-96d7-74b1672c5eb7.png)

@ilmartyrk did you have a solution for that? :)

**Comment by anettlinno** [comment]
> Also a grammar point, should be "A user" here.
> 
> But I also wonder if everyone wants to know every time someone votes. Especially if it's a big group.. Obviously they could turn it off, but then they would lose notifications for anything to do with voting, presumably, so including the vote ending, or being extended, or such like?
> 
> ![image](https://user-images.githubusercontent.com/61320526/173887251-ec982801-1050-4208-a65a-2d44a1586858.png)

Again, in MVP version we don´t go into the details of what kinds of notifications Users can order. We provide "all or nothing" solution and later we might change that (if we have more information).

@BeccaMelhuish  I think I wrote it somewhere but FYI that we will check all email texts after we have finished with the technical changes. Then we´ll go them all through and test them once in the end to save the time and constant changing work :)

**Comment by anettlinno** [comment]
> Also @ssin1901 and @anettlinno, what do you think if we include Kevin in user testing this too? Perhaps Sara and him have a discussion on a topic and he reviews the notifications too? As I am just giving my hunches here from a UX point of view, but he is our UX expert :)

Kevin has done all the UX designs for us and for the "My account" view we have asked new UX design :) We agreed with @ssin1901 that she will do the user tests and we´ll share the outcomes with Kevin anyway for the next iteration :)

**Comment by anettlinno** [comment]
> Also, (why I discovered the above) I was looking for the option to turn off all notifications that say "User X has joined this topic" as I thought if it was an organisation with 100 people for example, or more, I wouldn't want 100 joining notifications muddled up with all the more important/interesting/relevant things. But I couldn't see which tick box this would be. Maybe it should be a separate tick box on it's own, so you could just stop getting those but still get all the rest?

I think this is something we could try to explain in the Help Center -- which notification settings respond to which notifications. @ssin1901 what do you think? :)

**Comment by anettlinno** [comment]
> In notification emails, we're currently offering to **"Turn off notifications for this topic"** towards the end of the email.
> 
> User subscribes to receiving notifications because he/she considers them important, however after receiving so many notification emails recently, the button I'm really missing in place of the current one is: **"Receiving too many notifications? See ways to group them."**
> 
> And then offering periodical summaries - send notifications immediately; or group all notifications into an email once a day, once a week, etc. Wdyt?

Offering periodical summaries is very good idea but this is out of our MVP scope. Later should be definitely added as it would be great to receive an update on a daily bases.

In MVP version we thought to solve this as this (and let us know your thoughts @oksks :) )
User should have an opportunity to change topic notifications options from their notifications email and **turn them all off from one place.** For that we should have two options (links in email): 1) "Manage notification settings for this topic" (directs User to "Set topic notification" view) AND 2) Manage all notifications" (directs User to "My account" "Notifications" view

**Comment by BeccaMelhuish** [comment]
Ok super, thanks @anettlinno for the explanations! And great to hear there will be proper user tests too, I had understood that wasn't the case for some reason :)    

Re this one below, is it actually possible though, to turn off notifications for "User X joined topic X", without turning off other notifications? As if not, and if we wanted to allow users to do this without turning off other things at the same time, then it would be needed to be added as a separate tick box.. 

![image](https://user-images.githubusercontent.com/61320526/174845256-7300cc6f-50cf-4ad6-a85f-2c6b24f82911.png)

For this one below, I think it would be good if it could be noted for consideration when reviewing the email notification texts for the MVP. As even if it's not possible to choose whether you receive notifications that directly relate to your contributions or not, it would be good if at least the email could communicate this. As then if a user receives a message saying for example "X user has replied to **your** argument in X topic", then they would know they definitely (presumably) want to look at it. Whereas if they receive 20 messages all saying "X user has replied to **an** argument in X topic" then they wouldn't know if any of these are replies that relate to their own contribution or not.

![image](https://user-images.githubusercontent.com/61320526/174846201-a96cda23-5a80-4a24-9820-5bb1db7c9cb5.png)

**Comment by ilmartyrk** [comment]
Update:
- [x] Fixed broken translations
- [x] Removed notifications for users own actions
- [x] Updated e-mail subject and added general notifications link to e-mail
- [x] Grouped notifications about topic reports under Topic update notifications
- [x] Added notifications about comment reactions to argument notifications
- [x] Updated argument notification topic link in e-mail to highlight the argument notification is about
- [x] Removed all Join, Accept, Leave actions from notifications - @anettlinno @BeccaMelhuish (This can easily be put back, but thought maybe it is irrelevant information. specially in big topics. Also we could implement this in the periodical notifications (X number of users joined/left)

From a technical perspective, [RFC 2822](http://www.rfc-editor.org/info/rfc2822) states that the maximum number of characters in a subject line is 998 characters. However, a lot of email clients will impose a 255/256 character limit: https://www.theorchardagency.com.au/orchard_insights/subject-line-length/ - we already have  100 character limit for our title and seems that we don't have to implement any cutoff  for title in subject fields

**Comment by kevincrepin** [comment]
I think I covered everything, but please let me know if I missed something. There's a lot happening in this discussion :)

https://www.figma.com/file/KWTHiHwLZJ3ZFpNJfoVRXX/Platform-1.0?node-id=3320%3A39165

**My account** 
List of topics with on/off toggle and link which would open the dialog box. I added a search here because if I have a lot of topics I don't want to go through all the pages to find it.

**Dialog**
There's an option to toggle on/off the notifications. Additionally there are the different check options. If the user turns the notifications off, these checkboxes will be cleared. However the user can still check the boxes and by doing so the notifications will be turned on automatically. 

**Additionally**
- Kept the topic page here as well which has the "set topic notifications" action
- basically now the notifications tab in "my account" shows a full list of the topics I'm involved with. Which is kinda what the dashboard also does. We are duplicating here a bit and maybe it could be smarter. However the new dashboard will work a bit differently so in the redesign we will probably do this similarly. Anyways we can also add a "set topic notification" action in the dashboard topic view.


PS: About the dialog boxes with X closing etc. I agree and I prefer to not have an x to close at all when the user should make a conscious action. Either "save" or "cancel". Closing is too ambiguous. This is something we also take into the redesign.

**Comment by ilmartyrk** [comment]
Semi implemented  @kevincrepin will proceed after vacation

**Comment by BeccaMelhuish** [comment]
@ilmartyrk @anettlinno @ssin1901 I've spotted a UX issue when drafting up the notifications HC text (sorry @ssin1901 if I'm just repeating things you're flagging here):

The "toggle on/off" button and the "All actions" tick box seem to be linked the wrong way - when you toggle them on they all get unticked, and vice versa:

![image](https://user-images.githubusercontent.com/61320526/185615224-3829b947-81bf-4f8e-b195-3f451bc1eeb9.png)

![image](https://user-images.githubusercontent.com/61320526/185615278-1577d324-cebf-421c-a438-17ea3aff233f.png)

And when you tick and untick the tick box, nothing happens - the toggle stays wherever it was. So seems they're not linked in that direction. 

(I kind of wonder what's the point of having both these things actually - maybe only one or the other is needed?)

**Comment by BeccaMelhuish** [comment]
Also, was it decided to leave out an option to toggle on/off all notifications for all topics at once? As this would seem a nice easy way to allow people to manage their preferences nice and quickly, without having to go through each on separately.. But I guess it was too complicated for the MVP version?

**Comment by BeccaMelhuish** [comment]
Also I'd suggested that "topic" doesn't need to be in here twice :) 

![image](https://user-images.githubusercontent.com/61320526/185620738-7a4b322d-5c26-4416-87a2-c8432f928ea2.png)

**Comment by BeccaMelhuish** [comment]
@ilmartyrk @anettlinno @ssin1901 A bug I've found: when you go into the topic from here, and then click the browser back button once you are inside it (to go back to the manage notifications window) the back button doesn't work (at least not for me):

![image](https://user-images.githubusercontent.com/61320526/185623781-4363600d-d0da-4cce-a283-8eb69e8097a8.png)

And on this note, UX-wise I'd suggest not just having the hyperlinked topic title as a link to the topic itself, as you are in "Manage notifications" so would probably expect clicking on that to take you to managing notifications. I would suggest having a "View topic" button, next to the "Set topic notifications" button. Or next to the (no longer hyperlinked) heading itself perhaps. (The kind of thing that would come up in testing if it's a problem though (user testing I mean, not QA testing) - I can't remember what was decided about user testing for this feature? @ssin1901)

**Comment by anettlinno** [comment]
@ilmartyrk there are some issues with the notification system and I think we should prioritize fixing those :) 

Conclusion from Becca´s thread: 

- [ ] ON/OFF toggle is not working correctly (when clicking ANY notification ON, then the toggle should move to ON) and when opting all notifications out, toggle should move to OFF);
- [ ] "All notifications in this topic" and toggle are repeating each other and we could actually leave only one (I personally would prefer ON/OFF toggle for that) and ticks for separate actions;

> @ilmartyrk @anettlinno @ssin1901 A bug I've found: when you go into the topic from here, and then click the browser back button once you are inside it (to go back to the manage notifications window) the back button doesn't work (at least not for me):
> 
@BeccaMelhuish for me the "Back" button works nicely. @ssin1901 maybe you can test it too? 

> And on this note, UX-wise I'd suggest not just having the hyperlinked topic title as a link to the topic itself, as you are in "Manage notifications" so would probably expect clicking on that to take you to managing notifications. I would suggest having a "View topic" button, next to the "Set topic notifications" button. Or next to the (no longer hyperlinked) heading itself perhaps. (The kind of thing that would come up in testing if it's a problem though (user testing I mean, not QA testing) - I can't remember what was decided about user testing for this feature? @ssin1901)

I actually like that topics are hyperlinked -- this gives me an opportunity to quickly check what the topic is about (If I don´t remember) and the ON/OFF toggle and "Set notifications" is clearly seen and understandable. 

We agreed with @ssin1901 that we will do user test for this feature as well :)

**Comment by BeccaMelhuish** [comment]
@anettlinno Just to clarify for the last point, re the hyperlinks, I'm not suggesting there isn't a link (as I agree it's useful to be able to go into it to check it's the right one), but that it should be clearer what happens when you click on the link. As currently I think in the very short amount of time people will spend actually taking in the page (and quite possibly not initially noticing the "set notifications" link all the way over on the right hand side), they may well assume (like I did) that clicking on the link for that topic will take them into the notification settings for that topic (as they are in the Notifications management bit, would be a sensible assumption).

So I'm suggesting something like this instead:  

![image](https://user-images.githubusercontent.com/61320526/190851922-6425ccc9-f395-441e-ad03-8326f53caf41.png)

Or if you wanted to see if this really is a problem before (potentially) changing it, one way of testing this would be to show people a screenshot of this page/ask them to navigate to this page, and (without giving them much time to look at it) asking them their first instinct of what will happen when they click on the topic heading link (before they have actually done so). As otherwise the problem won't always come out in tests. For example, in a test: someone clicks on the link, looks at the topic, comes back to the notifications management tab - we wouldn't know what they were expecting to happen. Did they actually want to go into the topic and check it, or they were trying to get to the notifications settings. @ssin1901 just some thoughts from me if helpful for planning the test :)

**Comment by ssin1901** [comment]
@ilmartyrk ran a second round of tests and everything seems to be working as expected! Only a couple of quirky UX things

The notifications are displaying the topic titles inconsistently. It should either be topic or the actual title but it shows two different things.

<img width="727" alt="Screenshot 2022-11-27 at 00 25 13" src="https://user-images.githubusercontent.com/72747107/204111386-8815ca19-e8d6-4133-a9fe-ab2923d9de70.png">

<img width="743" alt="Screenshot 2022-11-27 at 00 28 35" src="https://user-images.githubusercontent.com/72747107/204111389-25aaf830-a60c-45de-a254-06d151e78645.png">

**Comment by anettlinno** [comment]
Triage 75. As for the MVP project current view works well and we´ll leave it as it is @BeccaMelhuish 

The topic headings will be reviewed and fixed.

Approximate fixing time 30 minutes.

**Comment by ilmartyrk** [comment]
Topic headings fixed, was a sync issue as already marked in issue https://github.com/citizenos/citizenos-fe/issues/1065

**Comment by anettlinno** [comment]
Closing the issue as the MVP version works well.
